File: src/main/java/com/soukhyatech/faceattendance/model/Employee.java
package com.soukhyatech.faceattendance.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.Instant;

@Entity
@Table(name = "employees", indexes = {
    @Index(name = "idx_emp_status_dept", columnList = "status, department"),
    @Index(name = "idx_emp_company", columnList = "company")
})
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Employee {

    @Id
    @Column(length = 20)
    @Pattern(regexp = "^EMP\\d{3,}$", message = "ID must match EMP### format")
    private String id;

    @Column(nullable = false, length = 100)
    @NotBlank
    @Size(min = 2, max = 100)
    private String name;

    @Column(nullable = false, length = 50)
    @NotBlank
    private String department;

    @Column(nullable = false, length = 100)
    @NotBlank
    private String role;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String descriptor;

    @Column(nullable = false, length = 64)
    private String descriptorHash;

    @Column(columnDefinition = "TEXT")
    private String image;

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private Instant createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at", nullable = false)
    private Instant updatedAt;

    @Column(name = "updated_by", length = 50)
    private String updatedBy;

    @Version
    @Column(nullable = false)
    @Builder.Default
    private Integer version = 1;

    @Column(nullable = false, length = 20)
    @Enumerated(EnumType.STRING)
    @Builder.Default
    private Status status = Status.Active;

    public enum Status {
        Active, Hibernate, On_Leave, Resigned
    }

    // Hibernate fields
    @Column(name = "hibernate_start_date")
    private String hibernateStartDate;

    @Column(name = "hibernate_end_date")
    private String hibernateEndDate;

    @Column(name = "hibernate_reason", length = 500)
    private String hibernateReason;

    // Enterprise HR fields
    @Column(length = 100)
    private String company;

    @Column(length = 100)
    private String designation;

    @Column(length = 10)
    private String gender;

    @Column(name = "date_of_joining")
    private String dateOfJoining;

    @Column(name = "date_of_confirmation")
    private String dateOfConfirmation;

    @Column(name = "last_working_day")
    private String lastWorkingDay;

    @Column(name = "aadhaar_number")
    private String aadhaarNumber;

    @Column(name = "pan_number")
    private String panNumber;

    @Column(name = "card_number")
    private String cardNumber;

    @Column(name = "phone_no")
    private String phoneNo;

    @Column(length = 100)
    @Email
    private String email;

    @Column(name = "reporting_to", length = 50)
    private String reportingTo;

    @Column(name = "device_code", length = 50)
    private String deviceCode;

    @Column(name = "sub_department", length = 100)
    private String subDepartment;

    @Column(length = 100)
    private String division;

    @Column(length = 50)
    private String grade;

    @Column(length = 100)
    private String team;

    @Column(length = 100)
    private String location;

    @Column(name = "employment_type", length = 20)
    private String employmentType;

    @Column(length = 50)
    private String category;

    @Column(name = "holiday_group", length = 100)
    private String holidayGroup;

    @Column(name = "shift_group", length = 100)
    private String shiftGroup;

    @Column(name = "shift_roster", length = 100)
    private String shiftRoster;

    @Column(length = 100)
    private String geofence;

    @Column(name = "device_expiry_rule_applicable")
    private Integer deviceExpiryRuleApplicable;

    @Column(name = "verification_type", length = 100)
    private String verificationType;

    @Column(name = "expiry_start_date")
    private String expiryStartDate;

    @Column(name = "expiry_end_date")
    private String expiryEndDate;
}

File: csharp/SoukhyaTech.FaceAttendance/Models/Employee.cs
using System.Text.Json.Serialization;

namespace SoukhyaTech.FaceAttendance.Models
{
    public class Employee
    {
        [JsonPropertyName("id")]
        public virtual string Id { get; set; } = string.Empty;

        [JsonPropertyName("name")]
        public virtual string Name { get; set; } = string.Empty;

        [JsonPropertyName("department")]
        public virtual string Department { get; set; } = string.Empty;

        [JsonPropertyName("role")]
        public virtual string Role { get; set; } = string.Empty;

        [JsonPropertyName("descriptor")]
        public virtual string Descriptor { get; set; } = string.Empty;

        [JsonPropertyName("image")]
        public virtual string? Image { get; set; }

        [JsonPropertyName("created_at")]
        public virtual string CreatedAt { get; set; } = DateTime.UtcNow.ToString("o");

        [JsonPropertyName("status")]
        public virtual string Status { get; set; } = "Active";

        [JsonPropertyName("hibernate_start_date")]
        public virtual string? HibernateStartDate { get; set; }

        [JsonPropertyName("hibernate_end_date")]
        public virtual string? HibernateEndDate { get; set; }

        [JsonPropertyName("hibernate_reason")]
        public virtual string? HibernateReason { get; set; }

        [JsonPropertyName("company")]
        public virtual string? Company { get; set; }

        [JsonPropertyName("designation")]
        public virtual string? Designation { get; set; }

        [JsonPropertyName("gender")]
        public virtual string? Gender { get; set; }

        [JsonPropertyName("date_of_joining")]
        public virtual string? DateOfJoining { get; set; }

        [JsonPropertyName("date_of_confirmation")]
        public virtual string? DateOfConfirmation { get; set; }

        [JsonPropertyName("last_working_day")]
        public virtual string? LastWorkingDay { get; set; }

        [JsonPropertyName("aadhaar_number")]
        public virtual string? AadhaarNumber { get; set; }

        [JsonPropertyName("pan_number")]
        public virtual string? PanNumber { get; set; }

        [JsonPropertyName("card_number")]
        public virtual string? CardNumber { get; set; }

        [JsonPropertyName("phone_no")]
        public virtual string? PhoneNo { get; set; }

        [JsonPropertyName("email")]
        public virtual string? Email { get; set; }

        [JsonPropertyName("reporting_to")]
        public virtual string? ReportingTo { get; set; }

        [JsonPropertyName("device_code")]
        public virtual string? DeviceCode { get; set; }

        [JsonPropertyName("sub_department")]
        public virtual string? SubDepartment { get; set; }

        [JsonPropertyName("division")]
        public virtual string? Division { get; set; }

        [JsonPropertyName("grade")]
        public virtual string? Grade { get; set; }

        [JsonPropertyName("team")]
        public virtual string? Team { get; set; }

        [JsonPropertyName("location")]
        public virtual string? Location { get; set; }

        [JsonPropertyName("employment_type")]
        public virtual string? EmploymentType { get; set; }

        [JsonPropertyName("category")]
        public virtual string? Category { get; set; }

        [JsonPropertyName("holiday_group")]
        public virtual string? HolidayGroup { get; set; }

        [JsonPropertyName("shift_group")]
        public virtual string? ShiftGroup { get; set; }

        [JsonPropertyName("shift_roster")]
        public virtual string? ShiftRoster { get; set; }

        [JsonPropertyName("geofence")]
        public virtual string? Geofence { get; set; }

        [JsonPropertyName("device_expiry_rule_applicable")]
        public virtual int? DeviceExpiryRuleApplicable { get; set; }

        [JsonPropertyName("verification_type")]
        public virtual string? VerificationType { get; set; }

        [JsonPropertyName("expiry_start_date")]
        public virtual string? ExpiryStartDate { get; set; }

        [JsonPropertyName("expiry_end_date")]
        public virtual string? ExpiryEndDate { get; set; }
    }
}
